api/v1beta2/awscluster_webhook.go

```diff
@@ -243,19 +251,37 @@ func (r *AWSCluster) validateNetwork() field.ErrorList {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("additionalControlPlaneIngressRules"), r.Spec.NetworkSpec.AdditionalControlPlaneIngressRules, "CIDR blocks and security group IDs or security group roles cannot be used together"))
 		}
 	}
+
 	return allErrs
 }
 
-func (r *AWSCluster) validateControlPlaneLBIngressRules() field.ErrorList {
+func (r *AWSCluster) validateControlPlaneLBs() field.ErrorList {
 	var allErrs field.ErrorList
 
-	if r.Spec.ControlPlaneLoadBalancer == nil {
-		return allErrs
+	// If the secondary is defined, check that the name is not empty and different from the primary.
+	if r.Spec.SecondaryControlPlaneLoadBalancer != nil {
+		if r.Spec.SecondaryControlPlaneLoadBalancer.Name == nil || *r.Spec.SecondaryControlPlaneLoadBalancer.Name == "" {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "secondaryControlPlaneLoadBalancer", "name"), r.Spec.SecondaryControlPlaneLoadBalancer.Name, "secondary controlPlaneLoadBalancer.name cannot be empty"))
+		}
```

Comment:
I'm guessing CAPI isn't using CEL validations for their API yet. This is an easy 1 liner in CEL on the secondary field
```
// +kubebuilder:validation:XValidation:rule="has(self.name) && self.name != ''",message="name is required for secondaryControlPlaneLoadBalancer"
```